src/components/structures/LeftPanel2.tsx

```diff
@@ -153,11 +153,16 @@ export default class LeftPanel2 extends React.Component<IProps, IState> {
                 header.style.width = `${headerStickyWidth}px`;
                 header.style.removeProperty("top");
                 gotBottom = true;
-            } else if ((slRect.top - (headerHeight / 3)) < top) {
+            } else if (((slRect.top - (headerHeight * 0.6) + headerHeight) < top) || sublist === sublists[0]) {
+                // the header should become sticky once it is 60% or less out of view at the top.
+                // We also add headerHeight because the sticky header is put above the scrollable area,
+                // into the padding of .mx_LeftPanel2_roomListWrapper,
+                // by subtracting headerHeight from the top below.
+                // We also always try to make the first sublist header sticky.
```

Comment:
won't this break the algorithm to scroll-to/collapse the sublist because that says

if is sticky: scroll to it
else expand/collapse it

Reply:
Good spot, I indeed can't collapse the first header now. Will fix.